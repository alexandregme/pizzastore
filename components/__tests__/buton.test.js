import React from 'react';
import { shallow } from 'enzyme';
import jest from 'jest-mock';
import expect from 'expect';
import Button from '../button';

describe("Button", ()=>{
  let mountedButton;

  beforeEach(()=>{
    mountedButton = shallow(<Button />);
  });

  it('renders without crashing', () => {
    shallow(<Button />);
    expect(mountedButton).toMatchSnapshot();
  });

  it('expect to have button tag with default value', () => {
    let button = mountedButton.find('button');
    expect(button).toHaveLength(1);
    expect(button.text()).toEqual('button');
  });

  it('call a function passed to it when clicked', ()=>{
    const mockCallBack = jest.fn();
    const mountedButtonWithCallback = shallow(<Button handleClick={mockCallBack} />);
    mountedButtonWithCallback.find('button').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});

describe("Button Custom component props", ()=>{
  let mountedButton;
  let props = {
    label: 'custom button'
  };

  beforeEach(()=>{
    mountedButton = shallow(<Button {...props} />);
  });

  it('expect to set the label custom button', () => {
    let button = mountedButton.find('button');
    expect(button).toHaveLength(1);
    expect(button.text()).toEqual('custom button');
  });
});
