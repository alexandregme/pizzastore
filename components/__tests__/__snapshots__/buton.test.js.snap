// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "button",
      "className": "btn btn-primary",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "button",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "button",
        "className": "btn btn-primary",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "button",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
